File: app/src/main/java/com/kota/Bahamut/ListPage/ListStateStore.java
package com.kota.Bahamut.ListPage;

import java.util.HashMap;
import java.util.Map;

public class ListStateStore {
    private static ListStateStore _instance = null;
    private final Map<String, ListState> _states = new HashMap();

    private ListStateStore() {
    }

    public static ListStateStore getInstance() {
        if (_instance == null) {
            _instance = new ListStateStore();
        }
        return _instance;
    }

    public ListState getState(String aBoardName) {
        ListState state = this._states.get(aBoardName);
        if (state != null) {
            return state;
        }
        ListState state2 = new ListState();
        this._states.put(aBoardName, state2);
        return state2;
    }
}

File: app/src/main/java/com/kota/Bahamut/ListPage/ListStateStore.kt
package com.kota.Bahamut.ListPage

class ListStateStore private constructor() {
    private val _states: MutableMap<String?, ListState?> = HashMap<Any?, Any?>()

    fun getState(aBoardName: String?): ListState {
        val state = this._states.get(aBoardName)
        if (state != null) {
            return state
        }
        val state2 = ListState()
        this._states.put(aBoardName, state2)
        return state2
    }

    companion object {
        private var _instance: ListStateStore? = null
        @JvmStatic
        val instance: ListStateStore
            get() {
                if (_instance == null) {
                    _instance = ListStateStore()
                }
                return _instance!!
            }
    }
}
